# Урок 7. Непараметрические тесты

1-4 задачи решать с помощью функций, 5ю вручную

Выберете тест и проверьте, есть ли различия между выборками:  
1 ) Даны две независимые выборки. Не соблюдается условие нормальности  
x1 380,420, 290  
y1 140,360,200,900  
Сделайте вывод по результатам, полученным с помощью функции

Так как даны две независимые выборки и не соблюдается условие нормальности выбираем непараметрический тест Манна-Уитни.

In [4]:
import numpy as np
from scipy import stats

In [5]:
x1 = np.array([380, 420, 290])
y1 = np.array([140, 360, 200, 900])

In [6]:
result_t1 = stats.mannwhitneyu(x1, y1)
result_t1

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Ответ:  
Так как pvalue > 0.05 (стандартное отклонение $ \alpha $) следовательно верна нулевая гипотеза нет статистически значимых различий.

2 ) Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут.  
Есть ли статистически значимые различия?  
1е измерение до приема препарата: 150, 160, 165, 145, 155  
2е измерение через 10 минут: 140, 155, 150, 130, 135  
3е измерение через 30 минут: 130, 130, 120, 130, 125

Так как мы тестируем множественные повторные измерения, то выбираем критерий Фридмана.

In [8]:
before = np.array([150, 160, 165, 145, 155])
measurement1 = np.array([140, 155, 150, 130, 135])
measurement2 = np.array([130, 130, 120, 130, 125])

In [11]:
result_t2 = stats.friedmanchisquare(before, measurement1, measurement2)
result_t2

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Ответ:  
Так как pvalue < 0.05 следовательно выбираем альтернативную гипотезу найдены статистически значимые различия.

3 ) Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут.  
Есть ли статистически значимые различия?  
1е измерение до приема препарата: 150, 160, 165, 145, 155  
2е измерение через 10 минут: 140, 155, 150, 130, 135  
3е измерение через 30 минут: 130, 130, 120, 130, 125  
Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

Так как сравниваем две группы и выборки зависимые, то выбираем критерий Уилкоксона.

In [12]:
before = np.array([150, 160, 165, 145, 155])
measurement1 = np.array([140, 155, 150, 130, 135])

In [16]:
result_t3 = stats.wilcoxon(before, measurement1)
result_t3

WilcoxonResult(statistic=0.0, pvalue=0.0625)

Ответ:  
Так как pvalue > 0.05 следовательно верна нулевая гипотеза нет статистически значимых различий.

4) Даны 3 группы учеников плавания.  
В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67  
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53  
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

In [17]:
group1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
group2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
group3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

In [18]:
result_t4 = stats.kruskal(group1, group2, group3)
result_t4

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

Ответ:  
Так как pvalue > 0.05 следовательно верна нулевая гипотеза нет статистически значимых различий.

5) Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%  
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

Так как неизвестна сигма генеральной совокупности используем критерий Стьюдента.  
$ t_H=\frac{\bar{X}-\mu}{\frac{\sigma_H}{\sqrt{n}}} $

$ \mu=2.5 $  
n = 10  
$ \alpha=5\%=0.05 $  
$ \bar{X}=(2.51+2.35+2.74+2.56+2.40+2.36+2.65+2.7+2.67+2.34)/10=2.5279999999999996 $

In [24]:
mu = 2.5
n = 10
x = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
X_mean = (2.51+2.35+2.74+2.56+2.40+2.36+2.65+2.7+2.67+2.34)/10
X_mean

2.5279999999999996

In [25]:
sigma = np.std(x, ddof=1)
sigma

0.1572542173961923

In [26]:
t_H = (X_mean - mu)/(sigma/np.sqrt(n))
t_H

0.5630613661802959

Найдём табличное значение критерия Стьюдента для выборки из 10 изделий и $ \alpha=0.05 $  
Число степеней свободы k = 10 - 1 = 9  
Вероятность p = 1 - 0.05 = 0.95  
Следовательно табличное значение критерия Стьюдента $ t_T=1.833 $

Проверим с помощью функции

In [32]:
stats.ttest_1samp(x, n)

Ttest_1sampResult(statistic=-150.2569474321155, pvalue=1.302122012238181e-16)

$ t_H=0.563 $  
$ t_T=1.833 $  
Ответ:  
Так как $ t_H<t_T $ следовательно подтверждается нулевая гипотеза нет статистически значимых различий.